pkg/bucket/bucket.go

```diff
@@ -29,33 +28,56 @@ const (
 )
 
 type Bucket struct {
-	buf []byte
-	src *[]byte
-
+	slots              [][]byte
 	init               bool
 	resyncOnNextPacket bool
 	step               int
 	headSN             uint16
+	initCapacity       int
 	maxSteps           int
 }
 
-func NewBucket(buf *[]byte) *Bucket {
+func NewBucket(capacity int) *Bucket {
```

Comment:
probably add `maxCapacity` as a safeguard?

Reply:
The bucket will not increase itself, all capacity is controlled by buffers.